conan/internal/rest/pkg_sign.py

```diff
@@ -1,46 +1,182 @@
+
+    def get_summary_file_path(self):
+        return os.path.join(self._signature_folder, self.SIGN_SUMMARY_FILENAME)
+
+    def is_pkg_signed(self):
+        try:
+            c = self.load_summary()
+        except FileNotFoundError:
+            return False
+        return bool(c.get("provider") and c.get("method"))
+
+    def create_summary_content(self):
+        """
+        Creates the summary content as a dictionary for manipulation
+        @return: Dictionary with the summary content
+        """
+        checksums = {}
+        for fname in os.listdir(self._artifacts_folder):
+            file_path = os.path.join(self._artifacts_folder, fname)
+            if os.path.isfile(file_path):
+                sha256 = sha256sum(file_path)
+                checksums[fname] = sha256
+        sorted_checksums = dict(sorted(checksums.items()))
+        content = copy.deepcopy(self.SIGN_SUMMARY_CONTENT)
+        content["files"] = sorted_checksums
+        return content
+
+    def load_summary(self):
+        """"
+        Loads the summary file from the signature folder
+        """
+        return json.loads(load(self.get_summary_file_path()))
+
+    def save_summary(self, content):
+        """
+        Saves the content of the summary to the signature folder using SIGN_SUMMARY_FILENAME as the
+        file name
+        @param content: Content of the summary file
+        @return:
+        """
+        assert content.get("provider")
+        assert content.get("method")
+        save(self.get_summary_file_path(), json.dumps(content))
 
 
 class PkgSignaturesPlugin:
     def __init__(self, cache, home_folder):
         self._cache = cache
-        signer = HomePaths(home_folder).sign_plugin_path
-        if os.path.isfile(signer):
-            mod, _ = load_python_file(signer)
-            # TODO: At the moment it requires both methods sign and verify, but that might be relaxed
-            self._plugin_sign_function = mod.sign
-            self._plugin_verify_function = mod.verify
-        else:
-            self._plugin_sign_function = self._plugin_verify_function = None
+        self.sign_plugin_path = HomePaths(home_folder).sign_plugin_path
+        self._plugin_sign_function = self._plugin_verify_function = None
+        self._output = ConanOutput(scope="[Package signing plugin]")
+        if os.path.isfile(self.sign_plugin_path):
+            mod, _ = load_python_file(self.sign_plugin_path)
+            try:
+                self._plugin_sign_function = mod.sign
+            except AttributeError:
+                pass
+            try:
+                self._plugin_verify_function = mod.verify
+            except AttributeError:
+                pass
 
-    def sign(self, upload_data):
+    def sign(self, upload_data, context="upload"):  # cache, upload,
+        results = {}
         if self._plugin_sign_function is None:
-            return
+            self._output.error(f"sign() function not found in {self.sign_plugin_path}")
```

Comment:
Why not raising an exception? If we are requesting to sign packages, and this returns without package signed because the plugin is not correct, better raise an exception, so the return code is error and the command fails, like in CI, otherwise, just an error message can be skipped, which is risky